test/components/preprocessors/test_document_splitter.py

```diff
@@ -141,3 +141,100 @@ def test_copy_metadata(self):
         for doc, split_doc in zip(documents, result["documents"]):
             assert doc.meta.items() <= split_doc.meta.items()
             assert split_doc.content == "Text."
+
+    def test_add_page_number_to_metadata_with_no_overlap(self):
```

Comment:
Let's break this into multiple unit tests. The comments you added already make it easier—the main reason is that it would be much easier to debug/adjust if something fails or changes.

Reply:
Okeey, later will update it.